src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1377,11 +1377,11 @@ def migrate_database(self) -> None:
                     SQLModel.metadata.create_all  # type: ignore[arg-type]
                 )
             with Session(self.engine) as session:
-                session.add(
-                    IdentitySchema(
-                        id=str(GlobalConfiguration().user_id).replace("-", "")
-                    )
+                id_ = (
+                    ServerConfiguration.get_server_config().external_server_id
+                    or GlobalConfiguration().user_id
                 )
```

Comment:
You missed the `replace("-", "")` part here. Is that intentional ? In the database, UUIDs are stored without their hyphens.

Reply:
Yeah that was intentional. The UUID is stored as a string without their hyphens, but sqlmodel/pydantic takes care of that for us (We use UUID in lots of other places). What really happened here I think is the following:
- We convert the UUID to a string and replace the hyphens
- We create the pydantic/SQLModel object, which converts the string to a UUID.
- When storing in the DB, SQLModel again converts to a string and replaces the hyphens 😄